Mathlib/Data/Polynomial/RingDivision.lean

```diff
@@ -394,6 +394,57 @@ theorem pow_rootMultiplicity_not_dvd {p : R[X]} (p0 : p ≠ 0) (a : R) :
     ¬(X - C a) ^ (rootMultiplicity a p + 1) ∣ p := by rw [← rootMultiplicity_le_iff p0]
 #align polynomial.pow_root_multiplicity_not_dvd Polynomial.pow_rootMultiplicity_not_dvd
 
+theorem Monic.mem_nonZeroDivisors {p : R[X]} (h : p.Monic) : p ∈ nonZeroDivisors R[X] :=
+  mem_nonZeroDivisors_iff.2 fun _ hx ↦ (mul_left_eq_zero_iff h).1 hx
+
+theorem mem_nonZeroDivisors_of_leadingCoeff_mem_nonZeroDivisors {p : R[X]}
```

Comment:
I think the name is too long ...
```suggestion
theorem mem_nonZeroDivisors_of_leadingCoeff {p : R[X]}
```
btw nonZeroDivisors has a scoped [notation](https://github.com/leanprover-community/mathlib4/blob/c6979569edc545f999b82d8a833b190c918aec2e/Mathlib/RingTheory/NonZeroDivisors.lean#L41)